model-optimizer/extensions/middle/ApplyPermutations.py

```diff
@@ -141,7 +141,10 @@ def permute_input_data(graph: Graph):
 
                 if permutation_data_node.has_and_set('permutation') and \
                         not is_input_data_in_correct_layout(node, in_port) and \
-                        len(port_to_check.data.get_shape()) >= 4:
+                        (len(port_to_check.data.get_shape()) >= 4 or \
+                            (node['type'] == 'StridedSlice' and node.in_port(in_port).data.get_shape()[0] >= 4)):
+                    # Kludge, 'it ain't much honest, but it works' since StridedSlice need to be permuted even inputs
```

Comment:
Please, avoid leaving such kind of comments

Reply:
removed, also moved this logic to general mechanisim